src/hotspot/share/opto/superword.cpp

```diff
@@ -2237,396 +1992,65 @@ void SuperWord::apply_memops_reordering_with_schedule(Node_List& memops_schedule
-        assert(cmp != nullptr, "must have cmp above CMove");
-        Node_List* cmp_pack = get_pack(cmp);
-        assert(cmp_pack != nullptr, "Bool must have matching Cmp pack");
-
-        Node* cmp_in1 = vector_opd(cmp_pack, 1);
-        Node* cmp_in2 = vector_opd(cmp_pack, 2);
-
-        Node* blend_in1 = vector_opd(p, 2);
-        Node* blend_in2 = vector_opd(p, 3);
-
-        VTransformBoolTest bool_test = _packset.get_bool_test(bool_pack);
-        BoolTest::mask test_mask = bool_test._mask;
-        if (bool_test._is_negated) {
-           // We can cancel out the negation by swapping the blend inputs.
-           swap(blend_in1, blend_in2);
-        }
-
-        // VectorMaskCmp
-        ConINode* test_mask_node  = igvn().intcon((int)test_mask);
-        BasicType bt = velt_basic_type(cmp);
-        const TypeVect* vt = TypeVect::make(bt, vlen);
-        VectorNode* mask = new VectorMaskCmpNode(test_mask, cmp_in1, cmp_in2, test_mask_node, vt);
-        phase()->register_new_node_with_ctrl_of(mask, p->at(0));
-        igvn()._worklist.push(mask);
-
-        // VectorBlend
-        vn = new VectorBlendNode(blend_in1, blend_in2, mask);
-      } else if (n->req() == 3) {
-        // Promote operands to vector
-        Node* in1 = nullptr;
-        bool node_isa_reduction = is_marked_reduction(n);
-        if (node_isa_reduction) {
-          // the input to the first reduction operation is retained
-          in1 = first->in(1);
-        } else {
-          in1 = vector_opd(p, 1);
-          if (in1 == nullptr) {
-            assert(false, "input in1 to vector operand was not created");
-            C->record_failure(C2Compiler::retry_no_superword());
-            return false; // bailout
-          }
-        }
-        Node* in2 = vector_opd(p, 2);
-        if (in2 == nullptr) {
-          assert(false, "input in2 to vector operand was not created");
-          C->record_failure(C2Compiler::retry_no_superword());
-          return false; // bailout
-        }
-        if (in1->Opcode() == Op_Replicate && (node_isa_reduction == false) && (n->is_Add() || n->is_Mul())) {
-          // Move invariant vector input into second position to avoid register spilling.
-          Node* tmp = in1;
-          in1 = in2;
-          in2 = tmp;
-        }
-        if (node_isa_reduction) {
-          const Type *arith_type = n->bottom_type();
-          vn = ReductionNode::make(opc, nullptr, in1, in2, arith_type->basic_type());
-          if (in2->is_Load()) {
-            vlen_in_bytes = in2->as_LoadVector()->memory_size();
-          } else {
-            vlen_in_bytes = in2->as_Vector()->length_in_bytes();
-          }
-        } else {
-          if (VectorNode::can_use_RShiftI_instead_of_URShiftI(n, velt_basic_type(n))) {
-            opc = Op_RShiftI;
-          }
-          vn = VectorNode::make(opc, in1, in2, vlen, velt_basic_type(n));
-          vlen_in_bytes = vn->as_Vector()->length_in_bytes();
-        }
```

Comment:
Note:
this is a sort of "catch all" case for the nodes with `req == 3`.

But I decided to spit away the reductions: `VTransformReductionVectorNode`.
All other cases are `VTransformElementWiseVectorNode`.